en/guide/writing-middleware.md

```diff
@@ -140,6 +142,82 @@ app.listen(3000)
 
 When you make a request to the root of the app, the app now displays the timestamp of your request in the browser.
 
+<h3>Middleware function validateCookies</h3>
+
+Finally, we'll create a middleware function that validates incoming cookies and sends a 400 response if cookies are invalid.
+
+Here's a simple function that validates the presence of certain cookies.
+
+```js
+function cookieValidator (cookies) {
```

Comment:
Should this show an actual async operation?  Just so we don't confuse beginners.

Reply:
yes